src/esp/core/logging.h

```diff
@@ -62,8 +284,8 @@ class LogMessageVoidify {
 #define ASSERT(x, ...)                                              \
   do {                                                              \
     if (!(x)) {                                                     \
-      LOG(ERROR) << "Assert failed: " #x << ", " << __FILE__ << ":" \
-                 << __LINE__;                                       \
+      ESP_ERROR() << "Assert failed: " #x << "," << __FILE__ << ":" \
+                  << __LINE__;                                      \
       exit(-1);                                                     \
```

Comment:
That reminds me, earlier we went and replaced most (all?) `ASSERT()`s with `CORRADE_ASSERT()` and/or `ESP_CHECK()`, what should we do with those? Should they stay or should they get reworked on top of this new logging system as well?

Personally I'd stay with the current implementation of `CORRADE_ASSERT()` / `ESP_CHECK()`. Since it's not really meant for logging, I think we don't need any of the features like subsystem prefix or output redirection -- conversely, we want it to *never* be redirected, and that's what the "vanilla" macros do. (Plus you'd have to rework all the `INTERNAL`/`UNREACHABLE`/... variants here, I don't think that's worth the time.)

Reply:
Yep, my thought was to stick with `CORRADE_ASSERT()` / `ESP_CHECK()` as is since those should never be silenced